go-controller/pkg/util/pool/pool_test.go

```diff
@@ -0,0 +1,124 @@
+package pool_test
+
+import (
+	"net"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util/pool"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("NetworkPool", func() {
+	const testNetwork = "test-network"
+	const ownerID1 = "namespace1/pod1"
+	const ownerID2 = "namespace2/pod2"
```

Comment:
is this how you plan on using the API in production ? could this raise any issue with duplicate pod names (not at the same time - something like a stateful set) ? 

do we want to add the dimension of the interface here ? e.g. pods w/ > 1 interface. 

I *think* we can keep it simple ... just thinking out loud.

Reply:
I meant that if `web-1` dies, a new pod with that name will be spun up. 

i.e. there are pods with the same name - just not at the exact same time. But, it might be possible to have a race in the API / OVN-K to react to a new pod being created when it last saw the "old" pod.

Reply:
pod names are not unique - they are unique at any point in time. but pod "abcd" at t=0 can be different than pod "abcd" at t=1. Right ? 

which means that your reconcile function must be prepared to handle that.

Reply:
> pod names are not unique - they are unique at any point in time. but pod "abcd" at t=0 can be different than pod "abcd" at t=1. Right ?

Do you mean in case the pod has been deleted and created?
If so, from what I see cluster-manager controller is single threaded, each pod is handled sequentially.
In a scenario a pod being deleted while being processed, the delete event will be handled after the create event is finished.
The mac will get allocated and then released.
WDYT?

Reply:
> Do you mean in case the pod has been deleted and created?

Yes ! Exactly that.

> In a scenario a pod being deleted while being processed, the delete event will be handled after the create event is finished.

I don't think it works like that - you do not need to receive the delete then the add operation. Most likely, this will be interpreted as an update.